zerver/lib/events.py

```diff
@@ -984,6 +984,11 @@ def post_process_state(user_profile: UserProfile, ret: Dict[str, Any],
 
         del ret['raw_users']
 
+    if user_profile is None:
+        ret['recent_private_conversations'] = []
+        del ret['raw_recent_private_conversations']
+        return
```

Comment:
I think this block is a noop given given that we're setting `raw_recent_private_conversations` properly and `notification_settings_null` should be True, so probably best to delete it for simpler control flow.